docs/merged_reduced_scans_table.md

```diff
@@ -0,0 +1,122 @@
+
+| Stage       | Explanation |
+| ----------- | ----------- |
+| setup       | The initial setup phase for the test (mustering resources, opening ports, etc.) |
+| dial        | The initial TCP dial connection to the remote |
+| tls         | The TLS handshake. `HTTPS` only. The [SNI header](https://en.wikipedia.org/wiki/Server_Name_Indication) containing the test domain is sent in this stage |
+| write       | Writing to the remote. For non-`HTTPS` tests this is where the domain is sent. |
+| read        | Reading from the remote |
+| http        | Verification of HTTP headers. `HTTP/S` only |
+| content     | Verification that the returned content matches the expected content. These are the most common types of errors and represent things like blockpages. |
+| final       | Reaching the final stage without encountering any problems in the previous stages. |
+| unknown     | The stage is not known. Usually these are new errors which should be investigated and classified  |
+
+#### Stages per Probe
+
+Not all tests include every stage depending on the type of test. For example since the Echo protocol does not involve TLS Echo tests will never fail with errors classified as `tls`. Here are the stages for each test type.
+
+##### Discard
+
+![discard connection diagram](diagrams/discard.svg)
+
+##### Echo
+
+![echo connection diagram](diagrams/echo.svg)
+
+##### HTTP
+
+![http connection diagram](diagrams/http.svg)
+
+##### HTTPS
+
+![https connection diagram](diagrams/https.svg)
+
+### Error Classes
+
+Basic errors represent simplest types of errors, as well as the `success` case (no error detected).
+
+Protocol errors are similar but not identical to the Network Error Logging standard's [Predefined Network Error Types](https://www.w3.org/TR/network-error-logging/#predefined-network-error-types). These errors may represent normal network failures and noise, but they may also expose interference by a middlebox to disrupt a connection. For example China and Iran are both known to use [TCP Reset Attacks](https://en.wikipedia.org/wiki/TCP_reset_attack) as a censorship method.
+
+Mismatch Errors are used when the connection is successful, but the content received does not match the content expected. This can happen in the case of blockpages, remote servers with unusual behavior, or complicated CDN networks serving many sites.
+
+| Error Class             | Explanation |
+| ----------------------- | ----------- |
+|                         |
+| **Basic Errors**        |
+|                         |
+| success                 | The test completed successfully and no interference was detected |
+| system                  | There was a test system failure, rendering the test invalid |
+| unknown                 | The class of the error is not known. Usually these are new errors which should be investigated and classified |
```

Comment:
I wonder if we need a "unknown" or whether we should output the error string as is. This way we know easily what the error is, and can write code to normalize it. I did that originally because it was a pain to figure out what the original message otherwise. And it helped me surface the errors that were not being handled.

Reply:
I disagree, currently there are 0 unknown errors, since the enum catches them all.

The point of unknown is to have a system for noticing if new errors show up. In that case we'll have to write some new classification code anyway, and it should just be an expansion of the case statement. Before we write that code I don't want a bunch of new errors blowing up the error selectors in the dashboards.

The errors is also still in the table, just in the `result` column.